modules/users/user.nix

```diff
@@ -65,10 +73,17 @@
     };
 
     shell = mkOption {
-      type = types.either types.shellPackage types.path;
-      default = "/sbin/nologin";
+      type = types.nullOr (types.either types.shellPackage types.path);
+      default = null;
       example = literalExpression "pkgs.bashInteractive";
-      description = "The user's shell.";
+      description = ''
+        The user's shell. This defaults to `null`.
+
+        When this is set to `null`, the value is managed by macOS instead of
+        `nix-darwin`. This means if the user has not been created yet,
+        `sysadminctl` will be called without the `-shell` flag which means the
+        user will have a default shell of `/bin/zsh`.
```

Comment:
This is scary: it means that a previously‐secure configuration with the `/sbin/nologin` default can now result in a system that inadvertently lets a user log in. Can we make sure we use `/sbin/nologin` by default on creation?

Reply:
Sure, I'll switch it to `/usr/bin/false` to match what macOS uses for its accounts

The end goal is the default will be determined by the type of user being created in a later PR